src/drivers/intel/hda/hda-dma.c

```diff
@@ -630,6 +631,24 @@ static int hda_dma_stop(struct dma_chan_data *channel)
 	return 0;
 }
 
+static int hda_dma_reset(struct dma_chan_data *channel)
+{
+	uint32_t val, flags;
+
+	irq_local_disable(flags);
+
+	dma_chan_reg_update_bits(channel, DGCS, DGCS_GEN, 0);
+
+	val = dma_chan_reg_read(channel, DGCS);
+
+	irq_local_enable(flags);
+
+	if (!(val & DGCS_GBUSY))
+		return 0;
+
+	return -EBUSY;
```

Comment:
We probably want to tell the user here in the logs

Reply:
ack, fixed now